src/main/java/org/minions/devfund/chelo/WhackAMole.java

```diff
@@ -0,0 +1,259 @@
+
+        // Printing the grid as first action to know where the moles were placed
+        whackAmole.printGrid();
+
+        int userInputX, userInputY;
+
+        // While the attempts left are greater than zero the user can submit coordinates
+        while (whackAmole.getAttemptsLeft() > 0) {
+            if (whackAmole.getMolesLeft() == 0) {
+                System.out.println("Whack-A-Mole CHAMP!!!");
+                break;
+            }
+
+            // Reading user input
+            System.out.println("Enter X: ");
+            userInputX = readUserInputForGame();
+            System.out.println("Enter Y: ");
+            userInputY = readUserInputForGame();
+
+            // Printing the coordinates submitted by the user
+            System.out.println("User input: [X=" + userInputX + "][Y=" + userInputY + "]");
+
+            // Conditional if the user wants to finish the game
+            if (userInputX == -1 || userInputY == -1) {
+                whackAmole.printGrid();
+                whackAmole.printStatus();
+                System.out.println("GAME OVER: User gave up");
+                break;
+            }
+
+            // Positioning the user's hit
+            whackAmole.whack(userInputX, userInputY);
+            whackAmole.printGridToUser();
+            whackAmole.printStatus();
+
+            numAttempts--;
+
+            if (numAttempts == 0) {
+                whackAmole.printGrid();
+                System.out.println("GAME OVER! NO MORE ATTEMPTS");
+                break;
+            }
+        }
+    }
+
+    /**
+     * A method to read user input and validate is a value between the range of array dimension.
+     * Validates a valid integer too.
+     * @return the integer value entered by the user.
+     */
+    public static int readUserInputForGame() {
+        Scanner scanner = new Scanner(System.in, "UTF-8");
+
+        String value = " ";
+        boolean flag = true;
+        while (flag) {
+            value = scanner.next();
+            if (!value.matches("^((-1)?|^[0-9])$")) {
+                flag = false;
+            } else {
+                System.out.print("Wrong input. Insert again: ");
+            }
+        }
+        return Integer.parseInt(value);
+    }
+
+    /**
+     * A method to read user input only integers.
+     * @return value submitted by the user
+     */
+    public static int readUserInputOnlyIntegers() {
+        Scanner scanner = new Scanner(System.in, "UTF-8");
+
+        String value = " ";
+
+        // The code to avoid Inner assignments should be avoided
+        boolean flag = true;
+        while (flag) {
+            value = scanner.next();
+            if (!value.matches("^[0-9]+$")) {
+                flag = false;
+            } else {
+                System.out.print("Wrong input. Insert again: ");
+            }
+        }
+
+        // The following code fails with error:
+        // Inner assignments should be avoided.
+//        String value;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)